white-test-app-backend/src/main/java/pl/lodz/p/white/whitetestapp/security/services/UserPrinciple.java

```diff
@@ -0,0 +1,98 @@
+package pl.lodz.p.white.whitetestapp.security.services;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+import pl.lodz.p.white.whitetestapp.model.Account;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+
+import static java.util.Arrays.asList;
+
+public class UserPrinciple implements UserDetails {
+    private static final long serialVersionUID = 1L;
+
+    private String username;
+
+    private String email;
+
+    @JsonIgnore
+    private String password;
+
+    private Collection<? extends GrantedAuthority> authorities;
+
+    public UserPrinciple(String username, String email, String password,
+                         Collection<? extends GrantedAuthority> authorities) {
+        this.username = username;
+        this.email = email;
+        this.password = password;
+        this.authorities = authorities;
+    }
+
+    public static UserPrinciple build(Account user) {
+        List<GrantedAuthority> authorities = asList(new SimpleGrantedAuthority(user.getRole().name()));
+
+        return new UserPrinciple(
+                user.getUsername(),
+                user.getEmail(),
+                user.getPasswordHash(),
+                authorities
+        );
+    }
+
+    public String getEmail() {
+        return email;
+    }
+
+    @Override
+    public String getUsername() {
+        return username;
+    }
+
+    @Override
+    public String getPassword() {
+        return password;
+    }
+
+    @Override
+    public Collection<? extends GrantedAuthority> getAuthorities() {
+        return authorities;
+    }
+
+    @Override
+    public boolean isAccountNonExpired() {
```

Comment:
takie coś powinno być?

Reply:
to jest nadpisane z nad-klasy, u nas nie ma obłsugi wygasnienia konta